addons/mail/static/src/utils/test_utils.js

```diff
@@ -169,47 +174,88 @@ function nextAnimationFrame() {
+         // exists, update its default value according to the one in the
+         // registry; If it does not exist, add it to the model definition.
+        const fieldNamesToFieldToInsert = fieldsToInsertRegistry.category(modelName).getEntries();
+        for (const [fname, fieldToInsert] of fieldNamesToFieldToInsert) {
+            if (fname in fields) {
+                fields[fname].default = fieldToInsert.default;
+            } else {
+                fields[fname] = fieldToInsert;
+            }
+        }
+        // apply default values for date like fields if none was passed.
+        for (const fname in fields) {
+            const field = fields[fname];
+            if (['date', 'datetime'].includes(field.type) && !field.default) {
+                const defaultFieldValue = field.type === 'date'
+                    ? () => moment().format('YYYY-MM-DD')
+                    : () => moment.utc().format("YYYY-MM-DD HH:mm:ss");
+                field.default = defaultFieldValue;
+            }
+        }
+    }
+    // add models present in the fake models registry to the model definitions.
+    const fakeModels = modelDefinitionsRegistry.category('fakeModels').getEntries();
+    for (const [modelName, fields] of fakeModels) {
+        modelDefinitions.set(modelName, fields);
+    }
+    return modelDefinitions;
+}
 
-    data.currentPartnerId = 3;
-    data['res.partner'].records.push({
-        display_name: "Your Company, Mitchell Admin",
-        id: data.currentPartnerId,
-        name: "Mitchell Admin",
-    });
-    data.currentUserId = 2;
-    data['res.users'].records.push({
-        display_name: "Your Company, Mitchell Admin",
-        id: data.currentUserId,
-        name: "Mitchell Admin",
-        partner_id: data.currentPartnerId,
-    });
+//------------------------------------------------------------------------------
+// Public: test lifecycle
+//------------------------------------------------------------------------------
 
-    data.publicPartnerId = 4;
-    data['res.partner'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicPartnerId,
-    });
-    data.publicUserId = 3;
-    data['res.users'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicUserId,
-        name: "Public user",
-        partner_id: data.publicPartnerId,
-    });
+async function beforeEach(self) {
+    const data = { ...TEST_USER_IDS };
+    const modelDefinitions = await modelDefinitionsPromise;
+    const recordsToInsertRegistry = registry.category('mail.model.definitions').category('recordsToInsert');
+    for (const [modelName, fields] of modelDefinitions) {
+        const records = [];
+        if (recordsToInsertRegistry.contains(modelName)) {
+            // avoid records corruption during the test.
+            records.push(...JSON.parse(JSON.stringify(recordsToInsertRegistry.get(modelName))));
```

Comment:
Not super clear what "records corruption" means.
I suppose you mean records are mutating somehow?

Reply:
Indeed, mock models used to return a function that generated records, rather than records directly, because those are modified in place in the mock server when there are updates to them.

Code after this PR is directly registering the records, not a generator for them, so I'm assuming that's why this "hack" is here. As I commented on the line, mock server should be improved in that regard instead, but probably out of scope.